### ISSUE TITLE
Missing plugin(native) entry in META's runtime package?

### ISSUE THREAD

**Issue by vprevosto** [ocaml-ppx/ppx_deriving#175]
I've been trying to dynamically load a `.cmxs` that depends on `ppx_deriving_yojson`, thus on `ppx_deriving`, using `findlib` to get all necessary dependencies loaded before my actual file. This failed when `findlib` said that `ppx_deriving_runtime.cmxa` should be loaded, instead of `ppx_deriving_runtime.cmxs`. Adding the line

```
plugin(native) = "ppx_deriving_runtime.cmxs"
```

as part of the `runtime` package description in `ppx_deriving`'s `META` file solved the issue.

Did I overlook something, or is this really an issue in `META`?

## steps to reproduce: 

Compile the following code with `ocamlfind ocamlopt -package findlib -package dynlink -linkpkg -linkall -o load load.ml` and launch `./load` (shortened from the mechanism used by Frama-C to load its plugins).

```ocaml
let () = Findlib.init ()

let () = Dynlink.init ()

let () = Dynlink.allow_unsafe_modules true

let predicates =
  ["plugin", ["native"]; "archive", ["plugin"]; "archive", ["native";"plugin"]]

let package = "ppx_deriving.runtime"

let find_file package acc (var, predicates) =
  match acc with
  | None ->
    (try Some (Findlib.package_property predicates package var)
     with Not_found -> None)
  | Some _ -> acc

let load package =
  let file = List.fold_left (find_file package) None predicates in
  match file with
  | Some file ->
    let base = Findlib.package_directory package in
    let full_path = Findlib.resolve_path ~base file in
    (try Dynlink.loadfile full_path
     with Dynlink.Error e ->
       Printf.eprintf "Dynlink error while loading %s: %s\n%!"
         full_path (Dynlink.error_message e))
  | None -> Printf.eprintf "No file found for loading %s\n%!" package

let () = List.iter load (Findlib.package_deep_ancestors ["native"] [package])
```

**Comment by whitequark** [comment]
I *very* vaguely recall some issue with native plugins that prevented this from working. But if it actually works, it's probably just a missing entry.

**Comment by vprevosto** [comment]
Thanks for your answer.

I can confirm that on my system (Arch Linux and opam 2.0), dynamic loading of `ppx_deriving_runtime` works (with the patched `META`, of course) as tested with the setting above, works from ocaml >= 4.03 (for which ppx_deriving 4.2.1 is not available anyway) up to 4.07. It can certainly be the case that there is some issue lying somewhere in a more exotic architecture. Nevertheless, I'd argue that adding the `plugin(native)` line in META will not harm this hypothetical architecture, while making it possible to use `Findlib` for dynamic loading of `ppx_deriving_runtime` whenever possible.

**Comment by vprevosto** [comment]
Done in #176 (I'm sorry if there was a possibility to open it directly here, I'm more familiar with gitlab processes than github ones).
